plugins/modules/swim_intent.py

```diff
@@ -768,18 +767,56 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
             params=site_params,
         )
         self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")
-        response = response['device'][0]['response']
-
-        if len(response) > 0:
-            for item in response:
-                if item["reachabilityStatus"] != "Reachable":
-                    self.log("""Reachability status of device '{0}' is '{1}', so cannot add it for distribution/activation
-                              task of swim image""".format(item["managementIpAddress"], item["reachabilityStatus"]), "INFO")
-                    continue
-                if "role" in item and (device_role is None or item["role"] == device_role.upper() or device_role.upper() == "ALL"):
-                    self.log("""Successfully fetched the device '{0}' associated with the given site '{1}' for SWIM distribution/activation
-                              task.""".format(item["managementIpAddress"], site_name))
-                    device_uuid_list.append(item["instanceUuid"])
+        response = response['device']
+
+        site_response_list = []
+        for item in response:
+            if item['response']:
+                for item_dict in item['response']:
+                    site_response_list.append(item_dict)
+
+        if device_role.upper() == 'ALL':
+            device_role = None
+
+        device_params = {
+            'series': device_series_name,
+            'family': device_family,
+            'role': device_role
+        }
+        device_list_response = self.dnac._exec(
+            family="devices",
+            function='get_device_list',
+            op_modifies=True,
+            params=device_params,
+        )
+
+        device_response = device_list_response.get('response')
+        if not response or not device_response:
+            self.log("Failed to retrieve devices associated with the site due to empty API response.")
```

Comment:
Can we print site name? What is the log level?

Reply:
added log level.